examples/crypto/SP800-108_counter_KDF/README.md

```diff
@@ -0,0 +1,122 @@
+<!--
+SPDX-FileCopyrightText: Copyright (c) 2023 Nordic Semiconductor ASA
+SPDX-License-Identifier: Apache-2.0
+-->
+
+NIST SP 800-108 counter-mode KDF
+================================
+
+This example shows a reference implementation for at key based key derivation function for counter mode that is conform to the [NIST SP 800-108 recommendation](https://csrc.nist.gov/pubs/sp/800/108/r1/final).
+
+It includes a HMAC and a CMAC variant.
+The CMAC variant implements the suggested additions to prevent a key control attack that is listed in Appendix B of [SP 800-108](https://csrc.nist.gov/pubs/sp/800/108/r1/final).
+
+
+
+Building the example
+********************
+To build the example CMake 3.10.2 or later is required.
+Also [mbedtls](https://github.com/Mbed-TLS/mbedtls) is required.
+You can either provide a mbedtls installation that exists or the build script will clone it into the build folder.
+Check the [requirements for building mbedtls](https://github.com/Mbed-TLS/mbedtls#compiling) to make sure all required dependencies are installed.
+
+
+To build the example then use CMake
+
+        [optional] export MBEDTLS_ROOT_PATH="/path/to/your/mbedtls"
+        mkdir build && cd build
+        cmake -G "Unix Makefiles" ..
+        cmake --build .
+
+
+Then run the sample
+
+        ./kbkdf
+        Deriving a key using SP800-108 HMAC counter mode...Done
+        Key:
+        ----------------------------------- len: 16 -----------------------------------
+        0x01 0x02 0x03 0x04 0x05 0x06 0x07 0x08 0x09 0x0a 0x0b 0x0c 0x0d 0x0e 0x0d 0x0f
```

Comment:
This is not correct. The 32 byte HMAC keys from main.c must be used instead.

Reply:
Oh yeah, thanks, die print out was not updated, fixed it